bin/guard

```diff
@@ -2,31 +2,114 @@
 
 require "pathname"
 
-default = Pathname(ENV.fetch("BUNDLE_GEMFILE", "Gemfile"))
-relative_to_binstub = Pathname(__FILE__).realpath + "../../Gemfile"
-gemfile = [default, relative_to_binstub].detect(&:exist?)
-
-if gemfile && !ENV["RUBYGEMS_GEMDEPS"]
-  ENV["BUNDLE_GEMFILE"] = gemfile.to_s
-  require "rubygems"
-  require "bundler/setup"
-else
-  require "rubygems"
-end
+class GuardReloader
+  class Config
+    def using_rubygems?
+      ENV["RUBYGEMS_GEMDEPS"]
```

Comment:
I think `disable-gems` works differently, it prevents automatic requirement of rubygems when starting ruby, compare (given rspec gem is installed):
```sh
ruby -e 'require "rubygems"; require "rspec"' # works even in pre 1.9 versions
ruby -e 'require "rspec"' # works starting with 1.9 as rubygems is required implicitly
ruby --disable-gems -e 'require "rspec"' # doesn't work as rubygems is not required
ruby --disable-gems -e 'require "rubygems"; require "rspec"' # works as rubygems is required explicitly
```